test/octalpermissions-success.yml

```diff
@@ -17,3 +17,13 @@
       file:
         path: foo
         mode: 0123
+
+    - name: octal permissions test success (4000)
+      file:
+        path: baz
+        mode: 4000
```

Comment:
Yes, it should be 04000. 4000 should trigger a warning. The leading 0 is always necessary to indicate octal format to the parser. 04000 in octal is 2048 in decimal, if a user wrote 4000, it may get interpreted as decimal and end up setting permission to be 4000 (which is bogus), when they intended it to be 2048.